lib/chef/resource.rb

```diff
@@ -1508,7 +1508,13 @@ def self.use(partial)
         dirname = ::File.dirname(partial)
         basename = ::File.basename(partial, ".rb")
         basename = basename[1..] if basename.start_with?("_")
-        class_eval IO.read(::File.expand_path("#{dirname}/_#{basename}.rb", ::File.dirname(caller_locations.first.path)))
+
+        # Support recursive `use`
+        callers = caller_locations
+        used_from = callers.first.label == "use" ? callers.at(3).path : callers.first.path
```

Comment:
This will not allow recursive, it'll allow a single extra level. I suspect what you want here is to walk `caller_locations` in reverse and find the *last* `use`, and use that path.